Additional Getting Started doc for Linux

carlescufi opened the pull request:
See individual commits

carlescufi committed:
doc: getting_started: Add CMake snap installation option

Document how snap can be used on Ubuntu to install a recent CMake
version.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
doc: getting_started: Document DTC version installation

Document that the SDK contains a bundled dtc binary and how to use it.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

jukkar commented on doc/getting_started/installation_linux.rst:
```diff
@@ -143,6 +143,12 @@ of obtaining a more recent one:
 You might also want to uninstall the CMake provided by your package manager to
 avoid conflicts.
 
+**A recent DTC version (1.5 or higher) is required**. Check what version you
```
I have older DTC (Fedora 30) and I can compile zephyr just fine
```
$ dtc --version
Version: DTC 1.4.7
```

carlescufi replied:
You are right, fixed now!

mbolivar replied:
@galak is DTC 1.4.7 officially supported?

galak replied:
The min version is 1.4.6, so anything newer works.

carlescufi committed:
doc: getting_started: Remove Linux distro versions

Remove the version numbers from the distribution listings, since those
get outdated quickly and on top of that they are causing confusion due
to those being now too old to contain recent versions of CMake and DTC.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

mbolivar reviewed (approved):
Looks good assuming the older DTC is OK

carlescufi committed:
doc: Consolidate Python and pip documentation

The documentation that describes how Python and pip interact with the OS
when installing packages used to be under a common section, and was
moved to the west bootstrap one later on. Since this information is
required early on (for example on Linux when installing CMake via pip3),
move the info to its own section and link to it from others.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
doc: Minor fixes to the Getting Started guide

A couple of minor fixes suggested by a user.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

mbolivar commented on doc/getting_started/index.rst:
```diff
@@ -24,6 +24,30 @@ Follow an operating system specific guide, then come back to this page.
 
+.. python-pip::
+
+Python and pip
```
From the commit log, if "(for example on Linux when installing CMake via pip3)" is important, why is this section placed after the linux specific document, instead of before?

carlescufi replied:
Good point, I will move it up. I think it doesn't hurt to have it at the top, since Zephyr relies so much on Python

mbolivar commented on doc/getting_started/index.rst:
```diff
@@ -24,6 +24,30 @@ Follow an operating system specific guide, then come back to this page.
    macOS <installation_mac.rst>
    Windows <installation_win.rst>
 
+.. python-pip::
```
missing `_`

dbkinder commented:
Recall you can see the generated doc artifacts here: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/17771/docs/getting_started/index.html

dbkinder reviewed (approved):
Thanks. These changes help. That said,  "Getting Started" with Zephyr is still daunting and with many opportunities to have things go wrong.  There's a TSC presentation coming up in a couple weeks (when Anas gets back) to talk about improving/simplifying the initial Getting Started developer experience.
